Sources/AsyncHTTPClient/ConnectionPool.swift

```diff
@@ -280,50 +290,65 @@ class HTTP1ConnectionProvider {
                 if connection.isActiveEstimation {
+                    logger.trace("leasing existing connection")
                     waiter.promise.succeed(connection)
                 } else {
+                    logger.trace("opening fresh connection (found matching but inactive connection)",
+                                 metadata: ["ahc-dead-connection": "\(connection)"])
                     self.makeChannel(preference: waiter.preference).whenComplete { result in
-                        self.connect(result, waiter: waiter, replacing: connection)
+                        self.connect(result, waiter: waiter, replacing: connection, logger: logger)
                     }
                 }
             }
         case .create(let waiter):
+            logger.trace("opening fresh connection (no connections to reuse available)")
             self.makeChannel(preference: waiter.preference).whenComplete { result in
-                self.connect(result, waiter: waiter)
+                self.connect(result, waiter: waiter, logger: logger)
             }
         case .replace(let connection, let waiter):
+            logger.trace("replacing \(connection), \(waiter)")
             connection.cancelIdleTimeout().flatMap {
                 connection.close()
             }.whenComplete { _ in
+                logger.trace("opening fresh connection (replacing \(connection), \(waiter)")
                 self.makeChannel(preference: waiter.preference).whenComplete { result in
-                    self.connect(result, waiter: waiter, replacing: connection)
+                    self.connect(result, waiter: waiter, replacing: connection, logger: logger)
                 }
             }
         case .park(let connection):
-            connection.setIdleTimeout(timeout: self.configuration.maximumAllowedIdleTimeInConnectionPool)
+            logger.trace("parking connection")
+            connection.setIdleTimeout(timeout: self.configuration.maximumAllowedIdleTimeInConnectionPool,
+                                      logger: logger.detachingRequestInformation(nil))
         case .closeProvider:
+            logger.debug("closing provider \(self)")
             self.closeAndDelete()
         case .none:
             break
         case .parkAnd(let connection, let action):
-            connection.setIdleTimeout(timeout: self.configuration.maximumAllowedIdleTimeInConnectionPool)
-            self.execute(action)
+            logger.trace("parking connection & doing further action",
+                         metadata: ["ahc-action": "\(action)"])
+            connection.setIdleTimeout(timeout: self.configuration.maximumAllowedIdleTimeInConnectionPool,
+                                      logger: logger.detachingRequestInformation(nil))
+            self.execute(action, logger: logger)
         case .closeAnd(let connection, let action):
+            logger.debug("closing \(connection), doing \(action)")
```

Comment:
I think we should be consistent with a log level. If `close`, `cancel` and `fail` are logged under `debug`, so should `create` and `replace`.

Reply:
It's inconsistent with "logging-design" (variables in log message).

Reply:
fixed both I think